README.md

```diff
@@ -59,6 +59,70 @@ f1.content = "The quick brown fox jumped over the lazy dog."
 bf1.save
 ```
 
+## Extending Directly Contained Derivatives
+
+Hydra Works comes with the ability to attach [Hydra Derivatives](https://github.com/projecthydra/hydra-derivatives) to a GenericFile. This allows a GenericFile of an image, video, audio file, or document type to directly contain appropriate derivative files such as thumbnail, extracted_text, mp3, ogg, mp4, and webm. The derivatives are configured in `lib/hydra/works/models/concerns/generic_file/contained_files.rb` and `lib/hydra/works/models/concerns/generic_file/derivatives.rb`.
+
+More directly contained derivatives can be configured for the Persist Derivatives service by extending `contained_files` and `makes_derivatives`, as long as the processor is defined in Hydra Derivatives.
+
+### contained_files
+```
+# lib/hydra/works/models/concerns/generic_file/contained_files.rb
```

Comment:
So the mechanism for adding custom derivatives is monkeypatching Hydra::Works modules in `lib/`?  I'm -0 on that pattern here, but I don't care enough about this to hold you up unless others also care.  I'd prefer to see Hydra::Works generate an initializer into an app with some commented-out code that can be used to customize things. Something like the following:

```ruby
# config/initializers/hydra_works.rb
Hydra::Works.configure
  # add_derivative(name: :new_derivative_name, uri: 'http://pcdm.org/use#ServiceFile', format: 'png', size: '400x400', thumbnail: { format: 'jpg', size: '338x493' }, mime_types: ['application/pdf'])
end
```

The cost of this approach is that it's going to require more code.  The benefits are providing a less troublesome upgrade path (if I have to copy/paste code into a monkeypatch, and the original code in Hydra::Works evolves, I need to track that manually) and abstracting away from users the details of how derivatives are generated (i.e., knowledge of `contained_files` and `makes _derivatives`).

I also don't love the API I sketched out above, so I'm happy for others to chime in. @terrellt @jcoyne @escowles @grosscol @elrayle @jpstroop etc.